packages/scene-composer/src/components/panels/scene-components/DataOverlayComponentEditor.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useCallback, useContext } from 'react';
+import { FormField, SpaceBetween, Textarea } from '@awsui/components-react';
+import { useIntl } from 'react-intl';
+
+import { IComponentEditorProps } from '../ComponentEditor';
+import { IDataOverlayComponentInternal, ISceneComponentInternal, useStore } from '../../../store';
+import { sceneComposerIdContext } from '../../../common/sceneComposerIdContext';
+import { Component } from '../../../models/SceneModels';
+
+import { DataBindingMapEditor } from './data-overlay/DataBindingMapEditor';
+
+export interface IDataOverlayComponentEditorProps extends IComponentEditorProps {
+  component: IDataOverlayComponentInternal;
+}
+
+export const DataOverlayComponentEditor: React.FC<IDataOverlayComponentEditorProps> = ({
+  node,
+  component,
+}: IDataOverlayComponentEditorProps) => {
+  const sceneComposerId = useContext(sceneComposerIdContext);
+  const updateComponentInternal = useStore(sceneComposerId)((state) => state.updateComponentInternal);
+  const valueDataBindingProvider = useStore(sceneComposerId)(
+    (state) => state.getEditorConfig().valueDataBindingProvider,
+  );
+  const { formatMessage } = useIntl();
+
+  const onUpdateCallback = useCallback(
+    (componentPartial: Partial<IDataOverlayComponentInternal>, replace?: boolean) => {
+      const componentPartialWithRef = { ref: component.ref, ...componentPartial };
+      updateComponentInternal(node.ref, componentPartialWithRef as ISceneComponentInternal, replace);
+    },
+    [node.ref, component.ref],
+  );
+
+  return (
+    <SpaceBetween size='s'>
+      <DataBindingMapEditor
+        valueDataBindingProvider={valueDataBindingProvider}
+        component={component}
+        onUpdateCallback={onUpdateCallback}
+      />
+
+      {component.dataRows.length > 0 &&
```

Comment:
What will it display if the length < 0 on the panel?

Reply:
it will just display a close button